discovery-provider/src/api/v1/users.py

```diff
@@ -525,20 +524,15 @@ class TrackHistoryFull(Resource):
     def get(self, user_id):
         """Fetch played tracks history for a user."""
         args = history_route_parser.parse_args()
-        decoded_id = decode_with_abort(user_id, ns)
         current_user_id = get_current_user_id(args)
-
         offset = format_offset(args)
         limit = format_limit(args)
-        get_tracks_args = {
-            "filter_deleted": False,
-            "user_id": decoded_id,
-            "current_user_id": current_user_id,
-            "limit": limit,
-            "offset": offset,
-            "with_users": True,
-        }
-        track_history = get_track_history(get_tracks_args)
+        get_tracks_args = GetUserListeningHistoryArgs(
```

Comment:
while I agree with your change (that we should be using current user id), that's not how this worked before and it makes this endpoint semantically kind of weird because you could specify `user_id` and it effectively does nothing.

can we keep that existing behavior and pass both current_user_id and user_id through to the queries file?

To get the behavior you added, we could just do
`if (user_id != current_user_id) return []` here

Reply:
Yup will do.